packages/runtime/datastore/src/channelStorageService.ts

```diff
@@ -48,7 +50,13 @@ export class ChannelStorageService implements IChannelStorageService {
         if (blob !== undefined) {
             return blob;
         }
-        return this.storage.readBlob(id);
+        const blobP = this.storage.readBlob(id);
+        blobP.catch((error) => {
+            this.logger.sendErrorEvent({ eventName: "ChannelStorageBlobError"}, error);
+            throw error;
```

Comment:
This will result in unhandled exceptions, because nobody observes the new promise created by blobP.catch() block.
So best way - remove throw from here, that's a dead-end promise fork (it's easier to think about promises as a tree, every time you add .then or .catch you create a child in that tree)

Alternatively, this function could return blobP.catch((eror) => {}...);
This way you would need to rethrow, as otherwise this function would return Promise<ArrayBufferLike | undefined> and eat exceptions.